dotcom-rendering/src/components/DiscussionApps.importable.tsx

```diff
@@ -19,8 +20,23 @@ const onReply = async (): Promise<CommentResponse> => {
 };
 
 const onRecommend = async (commentId: number): Promise<boolean> => {
-	return getdiscussionClient().recommend(commentId);
+	return getDiscussionClient()
+		.recommend(commentId.toString())
+		.then((discussionApiResponse) => {
+			if (
+				// eslint-disable-next-line no-underscore-dangle -- we don't have control over this name! It comes from the compiled Thrift models
+				discussionApiResponse.__type ===
+				DiscussionResponseType.DiscussionResponseWithResponse
+			) {
+				if (discussionApiResponse.response.statusCode === 200) {
+					return true;
+				}
+			}
+
+			return false;
```

Comment:
Is this the same as if we did a chained check?

```ts
return discussionApiResponse.__type ===
  DiscussionResponseType.DiscussionResponseWithResponse &&
  discussionApiResponse.response.statusCode === 200;
```

Reply:
hmm yes I think so, I'll rewrite!